use httr2 for newBiocPkgDOI

LiNk-NY opened the pull request:
Hi Andres, @jwokaty

I don't have a Datacite account for testing the re-written `generateBiocPkgDOI`
function. Could you review and make sure that the changes work?

You can also share the details via bitwarden.

Thank you!

LiNk-NY committed:
use httr2 for newBiocPkgDOI

LiNk-NY commented:
@jwokaty 
Note. This worked with when testing and generated a DOI at 

https://handle.test.datacite.org/10.82962/b9.bioc.multiassayexperiment

LiNk-NY commented:
Any updates on this @jwokaty ? 
Thanks!

jwokaty committed:
bump x.y.z version to odd y following creation of RELEASE_3_23 branch

jwokaty committed:
bump x.y.z version to even y prior to creation of RELEASE_3_23 branch

seandavi commented on R/newBiocPkgDOI.R:
```diff
@@ -76,19 +75,11 @@ generateBiocPkgDOI <- function(pkg, authors, pubyear, event = "publish", testing
-                         httr::add_headers(Authorization = paste("Basic", authorization, sep = " ")),
-                         httr::content_type("application/vnd.api+json"),
-                         encode = encode)
+  response <- request(base_url) |>
+      req_auth_basic(username, password) |>
+      req_headers("Content-Type" = "application/vnd.api+json") |>
+      req_body_json(payload) |>
+      req_perform()
```
@copilot apply changes based on [this feedback](https://github.com/seandavi/BiocPkgTools/pull/75#discussion_r3197445004)

LiNk-NY replied:
Thanks for taking a look. The change removed the `Content-Type` header from the request by _only_ using `req_body_json`. I have added it back in.

seandavi commented on R/newBiocPkgDOI.R:
```diff
@@ -76,19 +75,11 @@ generateBiocPkgDOI <- function(pkg, authors, pubyear, event = "publish", testing
-    response_json <- jsonlite::fromJSON(response_text)
-    return(response_json$data$id)
-  } else {
-    httr::message_for_status(response)
-  }
+  resp_body_json(response)$data$id
```
@copilot apply changes based on [this feedback](https://github.com/seandavi/BiocPkgTools/pull/75#discussion_r3197445020)

Copilot committed:
add resp_check_status in generateBiocPkgDOI

- Add resp_check_status() to surface HTTP errors before parsing body
- Update NAMESPACE to add resp_check_status import

Agent-Logs-Url: https://github.com/seandavi/BiocPkgTools/sessions/449565a2-c285-4206-8041-f299c3adb2c5

Co-authored-by: seandavi <92435+seandavi@users.noreply.github.com>

LiNk-NY committed:
add importFrom jsonlite fromJSON to pkgDownloadRank
